### ISSUE TITLE
Sweep starts at a sample count of 1 when beeing triggered by Impulse.ar(0)

### ISSUE THREAD

**Issue by dietcv** [supercollider/supercollider#6541]
## Environment
* SuperCollider version: 3.13.0
* Operating system: Windows 11

## Steps to reproduce
```
(
{ |t_trig = 1, duration = 0.02|
	
	var initTrigA = Impulse.ar(0);
	var oneShotPhaseA = Sweep.ar(initTrigA, SampleRate.ir) * (1 / duration * SampleDur.ir) * (PulseCount.ar(initTrigA) > 0);

	var initTrigB = Trig1.ar(t_trig, SampleDur.ir);
	var oneShotPhaseB = Sweep.ar(initTrigB, SampleRate.ir) * (1 / duration * SampleDur.ir) * (PulseCount.ar(initTrigB) > 0);
	
	[oneShotPhaseA, oneShotPhaseB];

}.plot(0.00041);
)
```
![grafik](https://github.com/user-attachments/assets/23edf3db-84cb-4f99-97f6-77f2d7c6b8c6)

## Expected vs. actual behavior
Sweep should always start from a sample count of 0.

**Comment by telephon** [comment]
To be added to the list: https://github.com/supercollider/rfcs/pull/19

**Comment by dietcv** [comment]
Thanks alot for having a look. Unfortunately i have no idea about how to handle the breaking change. 

Here is an additonal minor detail which is causing confusion imo. I think it would be appropriate to rename the `trig` argument of `Sweep` to `reset`. The argument name is misleading in the sense, that one needs a trigger to start the Ugen, when its actually a phase reset. This would also make it more aligned with `Duty`.

```
(
{
	var reset = Impulse.ar(100);
	var duty = Duty.ar(SampleDur.ir, reset, Dseries(0, 1));
	var sweep = Sweep.ar(reset, SampleRate.ir);
	[duty, sweep];
}.plot(0.041);
)
```

**Comment by elgiano** [comment]
Thanks for mentioning this, I agree it would be sensible to change it... and it would be another breaking change (even though a more visible one, affecting only sclang in cases when a user would call the argument by name, and in that case the error would be eloquent enough IMO).

I will bring this up to the next dev meeting to see if we prefer to schedule these breaking changes for 3.15 or if we want to do it in 3.14 already

**Comment by dietcv** [comment]
I hope i understand that correctly, but couldnt Sweep then just manually started one sample earlier like here with Phasor, to align the trigger with the wrap (compare both versions) ?

```
(
var rampToTrig = { |phase|
	var history = Delay1.ar(phase);
	var delta = phase - history;
	var sum = phase + history;
	var trig = (delta / sum).abs > 0.5;
	Trig1.ar(trig, SampleDur.ir);
};

{
	var phase, trig;
	//phase = Phasor.ar(DC.ar(0), 1000 * SampleDur.ir);
	phase = (Phasor.ar(DC.ar(0), 1000 * SampleDur.ir) - SampleDur.ir).wrap(0, 1);
	trig = rampToTrig.(phase);
	[phase, trig];
}.plot(0.0021).plotMode_(\plines);
)
```

**Comment by elgiano** [comment]
Thanks for the code example! I see in your example that Phasor needs a correction in order to trigger on sample 0, while Sweep doesn't.
If i understand you correctly, that's the bug: Sweep should behave like Phasor and not trig on sample 0 without a correction. The fact that Sweep's output doesn't start from 0 is due to the initialization bug we are talking about fixing... or did you mean something else?

**Comment by dietcv** [comment]
Currently when deriving a trigger from Phasor it will always be on the 1 sample and not on the 0 sample (initial trigger), because Phasor starts exactly at 0 and the delta calculation needs 1 sample delay. When we subtract the duration of 1 sample, the trigger will still be on the 1 sample.

Currently when deriving a trigger from Sweep it will always be on the 0 sample (initial trigger), because Sweep at sample 0 starts already at the sub-sample offset after the wrap from the pre-sample which is at 1, is this correct?

Actually im not sure which behaviour is favourable, but it should be consistant across the ugens imo which could be used to create a normalized ramp signal as a source of time (clock) and the output of Sweep should not differ for a reset with Impulse, Trig or Trig1.

**Comment by elgiano** [comment]
Correct, currently Sweep is one sample ahead of Phasor. And I agree it should be fixed, Sweep output should start from 0, like Phasor.

**Comment by dietcv** [comment]
Thanks alot :) This makes sense to me, then it would be consistant accross different attemps:

```
(
var rampToTrig = { |phase|
	var history = Delay1.ar(phase);
	var delta = phase - history;
	var sum = phase + history;
	var trig = (delta / sum).abs > 0.5;
	Trig1.ar(trig, SampleDur.ir);
};

{
	var phase, trig;
	phase = (Duty.ar(SampleDur.ir, DC.ar(0), Dseries(0, 1)) * 1000 * SampleDur.ir).wrap(0, 1);
	phase = (phase - SampleDur.ir).wrap(0, 1);
	trig = rampToTrig.(phase);
	[phase, trig];
}.plot(0.0021).plotMode_(\plines);
)
```
